docker-storage-setup.sh

```diff
@@ -931,49 +932,88 @@ create_lv() {
 return 0
 }
 
-setup_docker_root_volume() {
-  if [ -z "$DOCKER_ROOT_VOLUME_SIZE" ]; then
-    Fatal "Specify a valid value for DOCKER_ROOT_VOLUME_SIZE."
+setup_extra_volume() {
+  if [ -z "$3" ]; then
+    Fatal "Specify a valid value for EXTRA_VOLUME_SIZE."
   fi
 
-  if ! check_data_size_syntax $DOCKER_ROOT_VOLUME_SIZE; then
-    Fatal "DOCKER_ROOT_VOLUME_SIZE value $DOCKER_ROOT_VOLUME_SIZE is invalid."
+  if ! check_data_size_syntax $3; then
+    Fatal "EXTRA_VOLUME_SIZE value $3 is invalid."
   fi
 
-  if ! create_lv $DOCKER_ROOT_VOLUME_SIZE $DOCKER_ROOT_LV_NAME; then
-    Fatal "Failed to create volume $DOCKER_ROOT_VOLUME_SIZE of size ${DOCKER_ROOT_VOLUME_SIZE}."
+  if ! create_lv $3 $1; then
+    Fatal "Failed to create volume $1 of size ${3}."
   fi
 
-  if ! mkfs -t xfs /dev/$VG/$DOCKER_ROOT_LV_NAME > /dev/null; then
-    Fatal "Failed to create filesystem on /dev/$VG/${DOCKER_ROOT_LV_NAME}."
+  if ! mkfs -t xfs /dev/$VG/$1 > /dev/null; then
+    Fatal "Failed to create filesystem on /dev/$VG/${1}."
   fi
 
-  if ! mount_docker_root_volume; then
-    Fatal "Failed to mount docker root volume ${DOCKER_ROOT_LV_NAME} on ${DOCKER_ROOT_DIR}"
+  if ! mount_extra_volume $1 $2; then
+    Fatal "Failed to mount volume ${1} on ${2}"
   fi
 
   # setup right selinux label first time fs is created. Mount operation
   # changes the label of directory to reflect the label on root inode
   # of mounted fs.
-  if ! restore_selinux_context $DOCKER_ROOT_DIR; then
+  if ! restore_selinux_context $2; then
     return 1
   fi
 }
 
+setup_extra_lv_fs() {
+  [ -z "$EXTRA_VOLUME_MOUNT_DIR" ] && [ -z "$EXTRA_VOLUME_NAME" ] && return 0
+  local resolved_mount_dir_path=$RESOLVED_MOUNT_DIR_PATH
+  local extra_volume_name=$EXTRA_VOLUME_NAME
+  local extra_volume_size=$EXTRA_VOLUME_SIZE
+
```

Comment:
In this function you don't need local variables. You can simple pass global variables to function.

e.g.

setup_extra_dir $RESOLVED_MOUNT_DIR_PATH

Same for other variables